src/goFillStruct.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+'use strict';
+
+import vscode = require('vscode');
+import { byteOffsetAt, getBinPath, getFileArchive, getToolsEnvVars } from './util';
+import cp = require('child_process');
+import { promptForMissingTool } from './goInstallTools';
+import { outputChannel } from './goStatus';
+import { TextEdit } from 'vscode-languageclient/lib/main';
+
+// Interface for the output from fillstruct
+interface GoFillStructOutput {
+	start: number;
+	end: number;
+	code: string;
+}
+
+export function runFillStruct(editor: vscode.TextEditor) {
+	let args = getCommonArgs(editor);
+	if (!args) {
+		return;
+	}
+
+	return execFillStruct(editor, args);
+}
+
+function getCommonArgs(editor: vscode.TextEditor): string[] {
+	if (!editor) {
+		vscode.window.showInformationMessage('No editor is active.');
+		return;
+	}
+	if (!editor.document.fileName.endsWith('.go')) {
+		vscode.window.showInformationMessage('Current file is not a Go file.');
+		return;
+	}
+	let args = ['-modified', '-file', editor.document.fileName];
+	if (editor.selection.isEmpty) {
+		let offset = byteOffsetAt(editor.document, editor.selection.start);
+		args.push('-offset');
+		args.push(offset.toString());
+	} else if (editor.selection.start.line <= editor.selection.end.line) {
+		args.push('-line');
+		args.push(`${editor.selection.start.line + 1},${editor.selection.end.line + 1}`);
```

Comment:
When the command is run on a selection, `fillstruct` fails with `invalid value "..." for flag -line: strconv.ParseInt: parsing "...": invalid syntax
`
Looks like the `-line` flag takes a single number and we are passing 2 here.

Also, in what scenario do we expect the user select text and then run the command?

Reply:
I had that change sitting on my computer :) Since it takes a `-line`, I thought we might as well support selecting a whole line.